lib/cli.rb

```diff
@@ -0,0 +1,164 @@
+require_relative '../config/environment'
+require 'pry'
+require 'colorize'
+
+
+class CLI
+
+  
+
+  def create_user
+    puts `clear`
+    puts "Please type in your name...then press enter.. I guess..."
+    user_name = STDIN.gets.chomp.strip 
+    $user = User.find_or_create_by(name: user_name)
+  end 
+  
+  def greeting
+    puts `clear`
+    puts "Weather is breezy".colorize(:light_magenta)
+    puts 
+    puts "Weather is cool" 
+    puts 
+    puts "Don't get "+"washed".colorize(:light_magenta).italic+" away by too many choices."
+    puts 
+    puts "~ Select an option below "+"#{$user.name}".colorize(:cyan).italic+" ~"
+    puts 
+  end 
+
+  def cli_program
+    is_running = true 
+    while is_running 
+      puts "1. Search "+ "Weather".colorize(:cyan)
+      puts "2. View Search History"
+      puts "3. Delete History"
+      puts "4. Change a Search Record"
+      puts "5. Exit program"
+
+      choice = STDIN.gets.chomp 
+
+      if choice == "1"
+        puts `clear`
+        Search.all.each do |search|
+          puts "Location: #{search.search_location}"
+        end 
+        puts
+        puts "Please type in a City to see the "+"Weather".colorize(:cyan)+", then press enter" 
+        user_input = STDIN.gets.chomp.strip
+        
+        actual_location = Search.all.find_by(search_location: user_input)
+        
+
+          if actual_location
+            weather_description =  Weather.find_by(search_id: actual_location.id)
+            puts "The "+"Weather".colorize(:cyan)+" in #{user_input} is "+"#{weather_description.description}".colorize(:cyan)
+            puts
+            puts "Wasn't expecting that weather, now were you."
+            puts 
+            saved_location = Search.create(search_location: user_input)
+
+            saved_location_in_history = UsersSearch.create(search_id: saved_location.id, user_id: $user.id)
```

Comment:
History has no clear indication of meaning within this context. Instead name the variable based off of what object is being returned. In this case it is a UsersSearch instance that points to the row where search is matched with user.